src/pages/workspace/WorkspaceInviteMessagePage.tsx

```diff
@@ -1,139 +1,119 @@
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
 import SearchInputManager from './SearchInputManager';
-import {policyDefaultProps, policyPropTypes} from './withPolicy';
 import withPolicyAndFullscreenLoading from './withPolicyAndFullscreenLoading';
+import type {WithPolicyAndFullscreenLoadingProps} from './withPolicyAndFullscreenLoading';
 
-const personalDetailsPropTypes = PropTypes.shape({
-    /** The accountID of the person */
-    accountID: PropTypes.number.isRequired,
-
-    /** The login of the person (either email or phone number) */
-    login: PropTypes.string,
-
-    /** The URL of the person's avatar (there should already be a default avatar if
-  the person doesn't have their own avatar uploaded yet, except for anon users) */
-    avatar: PropTypes.string,
-
-    /** This is either the user's full name, or their login if full name is an empty string */
-    displayName: PropTypes.string,
-});
-
-const propTypes = {
+type WorkspaceInviteMessagePageOnyxProps = {
     /** All of the personal details for everyone */
-    allPersonalDetails: PropTypes.objectOf(personalDetailsPropTypes),
-
-    invitedEmailsToAccountIDsDraft: PropTypes.objectOf(PropTypes.number),
+    allPersonalDetails: OnyxEntry<PersonalDetailsList>;
 
-    /** URL Route params */
-    route: PropTypes.shape({
-        /** Params from the URL path */
-        params: PropTypes.shape({
-            /** policyID passed via route: /workspace/:policyID/invite-message */
-            policyID: PropTypes.string,
-        }),
-    }).isRequired,
+    /** An object containing the accountID for every invited user email */
+    invitedEmailsToAccountIDsDraft: OnyxEntry<InvitedEmailsToAccountIDs | undefined>;
 
-    ...policyPropTypes,
+    /** Updated workspace invite message */
+    workspaceInviteMessageDraft: OnyxEntry<string | undefined>;
 };
 
-const defaultProps = {
-    ...policyDefaultProps,
-    allPersonalDetails: {},
-    invitedEmailsToAccountIDsDraft: {},
-};
+type WorkspaceInviteMessagePageProps = WithPolicyAndFullscreenLoadingProps &
+    WorkspaceInviteMessagePageOnyxProps &
+    StackScreenProps<SettingsNavigatorParamList, typeof SCREENS.WORKSPACE.INVITE_MESSAGE>;
 
-function WorkspaceInviteMessagePage(props) {
+function WorkspaceInviteMessagePage({workspaceInviteMessageDraft, invitedEmailsToAccountIDsDraft, policy, route, allPersonalDetails}: WorkspaceInviteMessagePageProps) {
     const styles = useThemeStyles();
     const {translate} = useLocalize();
 
-    const [welcomeNote, setWelcomeNote] = useState();
+    const [welcomeNote, setWelcomeNote] = useState<string>();
 
     const {inputCallbackRef} = useAutoFocusInput();
 
     const getDefaultWelcomeNote = () =>
-        props.workspaceInviteMessageDraft ||
+        // workspaceInviteMessageDraft can be an empty string
+        // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+        workspaceInviteMessageDraft ||
```

Comment:
I think it was to keep the logic consistent with the way it was before the TS migration.
As you can see `props.workspaceInviteMessageDraft ||` was used, and since `workspaceInviteMessageDraft` can be the empty string the change to `??` would affect the logic before the TS migration, we wanted to escape any side updates when it's changes the logic.